src/main/java/org/xbill/DNS/SMIMEARecord.java

```diff
@@ -6,11 +6,12 @@
 import org.xbill.DNS.utils.*;
 
 /**
- * S/MIME cert association, draft-ietf-dane-smime.
+ * S/MIME cert association
```

Comment:
https://tools.ietf.org/html/rfc8162#section-2
```
   The SMIMEA wire format and presentation format are the same as for
   the TLSA record as described in Section 2.1 of [RFC6698].  The
   certificate usage field, the selector field, and the matching type
   field have the same format; the semantics are also the same except
   where RFC 6698 talks about TLS as the target protocol for the
   certificate information.
```

Looks like it's still basically identical to TLSARecord?  We should create a Base class to avoid the code duplication.

Reply:
Instead of a base class, passing the `Type` in a protected constructor to `TLSARecord` (like I've done for the CDS/CDNSKEY records) might make more sense. Less code duplication and no change of the TLSARecord inheritance chain.

Reply:
Logged issue #53 to create the tests to verify implementation is working as expected